ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/EmitCounters.java

```diff
@@ -0,0 +1,41 @@
+package com.mozilla.telemetry.contextualservices;
+
+import com.mozilla.telemetry.ingestion.core.Constant.Attribute;
+import com.mozilla.telemetry.metrics.PerDocTypeCounter;
+import com.mozilla.telemetry.transforms.PubsubConstraints;
+import java.util.Map;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+/**
+ * Extract reporting URL from document and filter out unknown URLs.
+ */
+public class EmitCounters
+    extends PTransform<PCollection<PubsubMessage>, PCollection<PubsubMessage>> {
+
+  public static EmitCounters of() {
+    return new EmitCounters();
+  }
+
+  private EmitCounters() {
+  }
+
+  @Override
+  public PCollection<PubsubMessage> expand(PCollection<PubsubMessage> messages) {
+    return messages.apply(
+        MapElements.into(TypeDescriptor.of(PubsubMessage.class)).via((PubsubMessage message) -> {
+          message = PubsubConstraints.ensureNonNull(message);
+          Map<String, String> attributes = message.getAttributeMap();
+
+          PerDocTypeCounter.inc(attributes, "valid_submission");
+          if (attributes.containsKey(Attribute.REQUEST_ID)) {
+            PerDocTypeCounter.inc(attributes, "valid_submission_merino");
```

Comment:
And we'll add this counter to the same graph that shows `valid_submission`, matching what we have from the BQ source.

Reply:
Do we have any outstanding documentation for the available counters we should update to reflect this addition?